### DISCUSSION TITLE
Could Avalonia provide the global hotkey service? that will work when the window is hidden.

### DISCUSSION THREAD

**Discussion by TheBetterSolution** [AvaloniaUI/Avalonia#16266] [Ideas]
Hi Avalonia team:

I think there are more applications need to handle the hotkey even if the window is hidden, because that's easy to use only the hotkey to do the work, that's dependent for the platform, for example, windows use the RegisterHotKey and WM_HOTKEY to provide this function, but implementation of MacOS/Linux is different.

Currently, we need to develop the code to implement it, please refer to:
https://github.com/AvaloniaUI/Avalonia/discussions/8823

Because this feature is good for the applications to improve user experiences, for example: bring up window from hidden state, or do common functions no need UI etc, could you please provide this feature?

Thanks very much.

**Comment by StefanKoell** [comment]
You may want to check out https://github.com/TolikPylypchuk/SharpHook

**Reply by TheBetterSolution** [reply]
Thanks, I will check this project.
